src/odemis/driver/xt_client.py

```diff
@@ -612,6 +624,144 @@ def set_autostigmator(self, channel_name, state):
+
+    def get_primary_stigmator(self):
+        """
+        Get the control values of the primary stigmator. Within the MBSEM system
+        there are two stigmators to correct for both beamlet astigmatism as well
+        as multi-probe shape. Each stigmator has two control values; x and y.
+
+        Returns
+        -------
+        tuple, (x, y) control values of primary stigmator, unitless.
+        """
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.get_primary_stigmator()
+
+    def set_primary_stigmator(self, x, y):
+        """
+        Set the control values of the primary stigmator. Within the MBSEM system
+        there are two stigmators to correct for both beamlet astigmatism as well
+        as multi-probe shape. Each stigmator has two control values; x and y.
+
+        Parameters
+        -------
+        (x, y) control values of primary stigmator, unitless.
+        """
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.set_primary_stigmator(x, y)
+
+    def primary_stigmator_info(self):
+        """"Returns the unit and range of the primary stigmator."""
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.primary_stigmator_info()
+
+    def get_secondary_stigmator(self):
+        """
+        Get the control values of the secondary stigmator. Within the MBSEM system
+        there are two stigmators to correct for both beamlet astigmatism as well
+        as multi-probe shape. Each stigmator has two control values; x and y.
+
+        Returns
+        -------
+        tuple, (x, y) control values of primary stigmator, unitless.
+        """
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.get_secondary_stigmator()
+
+    def set_secondary_stigmator(self, x, y):
+        """
+        Get the control values of the secondary stigmator. Within the MBSEM system
+        there are two stigmators to correct for both beamlet astigmatism as well
+        as multi-probe shape. Each stigmator has two control values; x and y.
+
+        Parameters
+        -------
+        (x, y) control values of primary stigmator, unitless.
+        """
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.set_secondary_stigmator(x, y)
+
+    def secondary_stigmator_info(self):
+        """"Returns the unit and range of the secondary stigmator."""
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.secondary_stigmator_info()
+
+    def get_dc_coils(self):
+        """
+        Get the four values of the dc coils.
+
+        Returns
+        -------
+        list, len 4
+            These 4 items describe 4x2 transformation matrix for 1 um beam shift using DC coils.
```

Comment:
I don't understand why these 4 values describe a 1um shift. I thought they are just the current position values of the beam shift and can be modified in any way.

Reply:
Actually, the doctring is not clear here. If I understand correctly, the return value is 4 tuples containing each 2 floats. So that's 8 values in total.

Reply:
It is indeed a list of 4 tuples each containing 2 floats.

I copied this explanation line from the document you linked to in the specification. Now that I look at it again, the 1 um doesn't make sense. Although I am also not sure what these values represent. Or is it good enough if I change this to:
`These 4 items describe 4x2 transformation matrix for a required beam shift using DC coils. `